src/core/Tree/Tree.tsx

```diff
@@ -0,0 +1,87 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { CommonProps, useTheme } from '../utils';
+import '@itwin/itwinui-css/css/tree.css';
+import cx from 'classnames';
+
+export const TreeContext = React.createContext<
+  | {
+      selectedNode?: React.ReactNode;
+      setSelectedNode?: (
+        node: React.ReactNode | ((prev: React.ReactNode) => React.ReactNode),
+      ) => void;
+      onNodeSelected?: () => void;
+      onNodeExpanded?: () => void;
+      checkbox?: React.ReactNode;
+      onNodeCheckboxSelected?: () => void;
+      nodeDepth?: number;
+    }
+  | undefined
+>(undefined);
+
+export type TreeProps = {
+  /**
+   * Callback fired when selecting a TreeNode.
+   */
+  onNodeSelected?: () => void;
+  /**
+   * Callback fired when expanding a TreeNode.
+   */
+  onNodeExpanded?: () => void;
+  /**
+   * Callback fired when checking a TreeNode's checkbox.
+   */
+  onNodeCheckboxSelected?: () => void;
+  /**
+   * Checkbox to be shown before TreeNodes, if undefined checkbox will not be shown.
+   * Recommended to use Checkbox component.
+   */
+  checkbox?: React.ReactNode;
```

Comment:
If we decide to move all callbacks to each node props then we might want to move checkbox too because that way users can handle all the checkbox callbacks and checked status themselves.

Reply:
So you would have to pass in checkbox and onNodeCheckboxSelected into each TreeNode?